### PR TITLE
Few interfaces added as mock test in app state

### PR DESCRIPTION
**Pull request by mdarade**
Mock test in app state to have async interfaces to
get, put blocks. One sync interface to get prevDigest
for given block.

### CODE REVIEW

File: bftengine/tests/bcstatetransfer/test_app_state.hpp [hunk 1]
```diff
@@ -77,16 +81,17 @@ class TestAppState : public IAppState {
     memcpy(&bl.block, block, blockSize);
     bl.actualSize = blockSize;
     last_block_ = blockId;
+    blocks_.emplace(blockId, bl);
```
Comment (glevkovich) [inline bftengine/tests/bcstatetransfer/test_app_state.hpp:84]: so what did it do until now?
the block was not really put anywhere?

File: bftengine/tests/bcstatetransfer/test_app_state.hpp [hunk 2]
```diff
@@ -77,16 +81,17 @@ class TestAppState : public IAppState {
...
 
-  // TODO - implement
   std::future<bool> putBlockAsync(uint64_t blockId,
                                   const char* block,
                                   const uint32_t blockSize,
                                   bool trylinkSTChainFrom = true) override {
-    ConcordAssert(false);
-    return std::async([]() { return false; });
+    std::future<bool> future =
+        std::async(std::launch::async, [&]() { return putBlock(blockId, block, blockSize, trylinkSTChainFrom); });
```
Comment (glevkovich) [inline bftengine/tests/bcstatetransfer/test_app_state.hpp:93]: Can you please rename all trylinkSTChainFrom to lastBlock? (only in this file)
This is a recent change, too bad I didn't tell you about it.

**Review by glevkovich — APPROVED** [review]

**Review by glevkovich — APPROVED** [review]
LGTM!
one small comment and a question.

File: bftengine/tests/bcstatetransfer/test_app_state.hpp (hunk 1 above)
Comment (mdarade) [reply bftengine/tests/bcstatetransfer/test_app_state.hpp:84]: Yes, block was not written anywhere.

**Review by f-squirrel — APPROVED** [review]
LGTM!

**Pull request merged**